server/routes/createTag.ts

```diff
@@ -0,0 +1,29 @@
+import lookupCommunityIsVisibleToUser from '../util/lookupCommunityIsVisibleToUser';
+import { Response, NextFunction } from 'express';
+import { UserRequest } from '../types';
+
+const createTag = async (models, req: UserRequest, res: Response, next: NextFunction) => {
+  console.dir(req.body);
+  const [chain, community] = await lookupCommunityIsVisibleToUser(models, req.body, req.user, next);
+  if (!chain && !community) return next(new Error('Invalid chain or community'));
+  if (chain && community) return next(new Error('Invalid chain or community'));
+  if (!req.user) return next(new Error('Not logged in'));
+  if (!req.body.name) return next(new Error('Tag name required'));
```

Comment:
Should we add a check here (and on front end) that restricts tags to only admins? @raykyri

Reply:
@dillchen I'm going to add that check right now, that's a good one. Users can create tags when they're directly associated with a thread they're authoring. This is admin only.